src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -627,6 +627,10 @@ Block InterpreterSelectQuery::getSampleBlockImpl()
         if (!analysis_result.need_aggregate)
         {
             // What's the difference with selected_columns?
```

Comment:
It is the header we want from remote server, to execute query up to WithMergeableState
When there is an ORDER BY, it is executed on remote server firstly, then we execute merge sort on initiator.
To execute ORDER BY, we need to calc ORDER BY keys.

Reply:
Added a comment.